src/neptune/handler.py

```diff
@@ -736,6 +736,20 @@ def __delitem__(self, path) -> None:
     @feature_temporarily_unavailable
     @check_protected_paths
     def pop(self, path: str = None, *, wait: bool = False) -> None:
+        """Completely removes the namespace and all associated metadata stored under the path.
+
+        Args:
+            path: Path of the namespace to be removed.
+            wait: By default, logged metadata is sent to the server in the background.
```

Comment:
`wait` seems not to express the effect properly in the context of `pop`. We will wait to all metadata to be removed :wink:

Reply:
Isn't it technically the case that with `wait=True`, any queued operations (which may include earlier logging calls) are completed before continuing? And not just this particular `pop()` operation.

Conversely, if `wait` is used in some call after `pop()`, do we not have the same phrasing problem?

```
run["huge_field"].pop()
run["a"].assign("aaa", wait=True)
```

I.e. in this case we'll technically wait for deletion rather than sending of metadata. Just wondering if we should adjust this wording globally.

Reply:
From a technical point of view, you're absolutely right 😉 . I've started to feel that it looks like being copy-pasted and doesn't match the actual use-case but I'm good as long as it's ok for you.

Reply:
Thanks for confirming 😃 We'll think of some more accurate and helpful phrasing in a separate update.